strings/boyer_moore.cpp

```diff
@@ -0,0 +1,166 @@
+/**
+ * @file
+ * @brief Boyer-moore's algorithm finding all occurrences of pattern in given
+ * text.
+ * @author [Stoycho Kyosev](https://github.com/stoychoX)
+ */
+#include <cassert>
+#include <cstring>
+#include <iostream>
+#include <limits>
+#include <vector>
+
+#define APLHABET_SIZE CHAR_MAX
+
+/// @brief A structure representing all the data we need to search the
+/// preprocessed pattern in text.
```

Comment:
Please document the functions/anything else similar to this format.
```suggestion
/**
 * @brief A structure representing all the data we need to search the
 * preprocessed pattern in the text.
```